src/lib/config.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    env,
+    fs::{canonicalize, read_to_string},
+    io::{stderr, Write},
+    path::Path,
+    process,
+};
+
+use getopts::Options;
+
+pub struct Config {
+    /// The maximum number of parallel jobs to run.
+    pub maxjobs: usize,
+    /// The port to listen on.
+    pub port: u16,
+    /// A *fully canonicalised* path to the directory containing per-repo programs.
+    pub reposdir: String,
+    /// The GitHub secret used to validate requests.
+    pub secret: String,
+    /// An optional email address to send errors to.
+    pub email: Option<String>,
```

Comment:
Should this be a vector to allow multiple recipients?

Reply:
This is a temporary feature which will be expanded in the future, so it's OK for now I think.